google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProviderTest.java

```diff
@@ -392,33 +390,34 @@ public void testExists_trailingSlash() {
   }
 
   @Test
-  public void testExists_trailingSlash_disablePseudoDirectories() throws IOException  {
+  public void testExists_trailingSlash_disablePseudoDirectories() throws Exception  {
     try (CloudStorageFileSystem fs = forBucket("military", usePseudoDirectories(false))) {
       assertThat(Files.exists(fs.getPath("fashion/"))).isFalse();
     }
   }
 
   @Test
-  public void testDelete() throws IOException {
+  public void testDelete() throws Exception {
     Files.write(Paths.get(URI.create("gs://love/fashion")), "(✿◕ ‿◕ )ノ".getBytes(UTF_8));
     assertThat(Files.exists(Paths.get(URI.create("gs://love/fashion")))).isTrue();
     Files.delete(Paths.get(URI.create("gs://love/fashion")));
     assertThat(Files.exists(Paths.get(URI.create("gs://love/fashion")))).isFalse();
   }
 
   @Test
-  public void testDelete_dotDirNotNormalized_throwsIae() throws IOException {
+  public void testDelete_dotDirNotNormalized_throwsIae() throws Exception {
     thrown.expect(IllegalArgumentException.class);
     Files.delete(Paths.get(URI.create("gs://love/fly/../passion")));
   }
 
   @Test
-  public void testDelete_trailingSlash() throws IOException {
+  public void testDelete_trailingSlash() throws Exception {
+    thrown.expect(CloudStoragePseudoDirectoryException.class);
```

Comment:
You are right. Now that the other code was merged, it's actually OK to delete a non-existent folder. Fixed. I also added a test for the case where pseudo-directories are disabled, for good measure.